File: java/air-io-network/src/main/java/org/feuyeux/air/io/network/netty/udp/cmd2/codec/ControlInfoCodecFactory.java
package org.feuyeux.air.io.network.netty.udp.cmd2.codec;

import org.feuyeux.air.io.network.netty.udp.cmd2.entity.CommandType;

/**
 * Created by erichan on 2/10/14.
 */
public final class ControlInfoCodecFactory {
    private ControlInfoCodecFactory() {
    }

    public static ControlInfoCodec getCodec(int type) {
        CommandType commandType = CommandType.getInstance(type);
        return getCodec(commandType);
    }

    public static ControlInfoCodec getCodec(CommandType commandType) {
        switch (commandType) {
            case KEY: {
                return new KeyControlInfoCodec();
            }
            default: {
                return null;
            }
        }
    }
}

File: ios/learning/UdpCmd/UdpCmd/ControlInfoCodecFactory.m
//
//  ControlInfoCodecFactory.m
//  UdpCmd
//
//  Created by feuyeux@gmail.com on 14-2-13.
//  Copyright (c) 2014年 feuyeux@gmail.com. All rights reserved.
//

#import "ControlInfoCodecFactory.h"

@implementation ControlInfoCodecFactory
+ (id<ControlInfoCodec>) getInstance:(CommandType)commandType{
    switch (commandType){
        case KEY:
            return [[KeyControlInfoCodec  alloc]  init];
        case Mouse:
            return NULL;
    }
    return NULL;
}

@end
